src/experiment_handler.py

```diff
@@ -14,22 +14,25 @@
     :copyright: 2018 SuperDARN Canada
     :author: Marci Detwiller
 """
-
-import zmq
-import sys
 import argparse
-import inspect
 import importlib
-import threading
+import inspect
+from pathlib import Path
 import pickle
+import sys
+import threading
+
+import structlog
+import zmq
+
 from utils.options import Options
 from utils import socket_operations
 from experiment_prototype.experiment_exception import ExperimentException
 from experiment_prototype.experiment_prototype import ExperimentPrototype
 
-from utils import log_config
-log = log_config.log(log_level='NOTSET', console=False, logfile=False, aggregator=False)
-
+caller = Path(inspect.stack()[-1].filename)
+module_name = caller.name.split('.')[0]
+log = structlog.getLogger(module_name)
```

Comment:
So I believe this block is required for import of experiment_handler to work correctly when imported (like by experiment_unittests.py), otherwise `log` is undefined in the functions because the `if __name__ == '__main__'` block is not run. In other modules like `data_write.py`, they are never imported, so the above three lines aren't present. What would probably be better would be to take the above three lines, and put them in an `else` after `if __name__ == '__main__'`. This may fix the double-logging of experiment_handler talked about in #419, I'll have to play around with it and confirm.